mypy/semanal.py

```diff
@@ -770,6 +770,8 @@ def analyze_base_classes(self, defn: ClassDef) -> None:
                     self.fail("Cannot subclass NewType", defn)
                 base_types.append(base)
             elif isinstance(base, AnyType):
+                if self.options.disallow_subclassing_any:
```

Comment:
@gvanrossum -- whoops, I must have overlooked this comment! I've pushed a new diff that reports the name when possible. Though, as you suspected, finding the module the base class was imported from was somewhat tricky.

If you'd like, I can make the error message even more descriptive and leave a brief note explaining possible causes of this error and potential fixes.